process stops after "Symbol" for page #11/12 SheetStub 1412 | Error in performing [LINKS] .... because "pitches" is null

**Describe the bug**
I have an orchestral partiture with 12 pages.
Only on page 11 I get above error, all other pages perform ... we.. reasonalby with a couple of quirks.

Process stops after "Symbols", and I cannot generate any output file.

Any cue on a workaround how to identify the offending part, so that I might clean / deassign it to finish processing would be welcome.

**To Reproduce**
If possible and relevant, post the input image so that we can reproduce the problem.

I don't own the right to publish the score.
I can mail it to sbdy on promise not to publish it.

**Expected behavior**
Processing finishing, may be with some quirks in the resulting xml that can be manually corrected

**Screenshots**

This ist what I see on the command line I start audiveris from
```
INFO  [We Shall Overcome - Orchester - Partitur - SL Edit 1_00a#11]                  StepMenu 224  | End of sheet step SYMBOLS
INFO  [We Shall Overcome - Orchester - Partitur - SL Edit 1_00a#11]                 SheetStub 1481 | Sheet#11 reset to binary.
INFO  [We Shall Overcome - Orchester - Partitur - SL Edit 1_00a#11]              ScaleBuilder 241  | Beam  guessed height:  4 -- 0.50 of 9 interline
INFO  [We Shall Overcome - Orchester - Partitur - SL Edit 1_00a#11]              ScaleBuilder 264  | Beam measured height:  6 -- 0.50 of [3..9] range at 248% of needed quorum
INFO  [We Shall Overcome - Orchester - Partitur - SL Edit 1_00a#11]                 ScaleStep 65   | Scale{ small_interline(7,8,9) interline(11,11,12) line(1,1,2) beam(6)}
INFO  [We Shall Overcome - Orchester - Partitur - SL Edit 1_00a#11]            LinesRetriever 1500 | Global slope: 0.00000
INFO  [We Shall Overcome - Orchester - Partitur - SL Edit 1_00a#11]         ClustersRetriever 326  | Retrieved raw line clusters: 20 of sizes [5] with interline(11,11,12)
INFO  [We Shall Overcome - Orchester - Partitur - SL Edit 1_00a#11]            LinesRetriever 1516 | Searching clusters with smallInterline: 8
INFO  [We Shall Overcome - Orchester - Partitur - SL Edit 1_00a#11]         ClustersRetriever 326  | Retrieved raw line clusters: 4 of sizes [5] with interline(7,8,9)
INFO  [We Shall Overcome - Orchester - Partitur - SL Edit 1_00a#11]                 PeakGraph 310  | Systems: #1[1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24]
INFO  [We Shall Overcome - Orchester - Partitur - SL Edit 1_00a#11]             BarsRetriever 1304 | System#1
INFO  [We Shall Overcome - Orchester - Partitur - SL Edit 1_00a#11]             BarsRetriever 1307 |    PartGroup{bracket number:1 barline:true staves:1-4}
INFO  [We Shall Overcome - Orchester - Partitur - SL Edit 1_00a#11]             BarsRetriever 1307 |    PartGroup{bracket number:1 barline:true staves:5-6}
INFO  [We Shall Overcome - Orchester - Partitur - SL Edit 1_00a#11]             BarsRetriever 1307 |    PartGroup{bracket number:1 barline:true staves:7-11}
INFO  [We Shall Overcome - Orchester - Partitur - SL Edit 1_00a#11]             BarsRetriever 1307 |    PartGroup{bracket number:1 barline:false staves:12}
INFO  [We Shall Overcome - Orchester - Partitur - SL Edit 1_00a#11]             BarsRetriever 1307 |    PartGroup{bracket number:1 barline:false staves:13-14}
INFO  [We Shall Overcome - Orchester - Partitur - SL Edit 1_00a#11]             BarsRetriever 1307 |    PartGroup{bracket number:1 barline:true staves:15-19}
INFO  [We Shall Overcome - Orchester - Partitur - SL Edit 1_00a#11]             SystemManager 779  | 23 parts along 1 system
INFO  [We Shall Overcome - Orchester - Partitur - SL Edit 1_00a#11]             StemSeedsStep 86   | stem(2 max:2)
INFO  [We Shall Overcome - Orchester - Partitur - SL Edit 1_00a#11]             HeadSeedTally 235  | Scale information: HeadSeeds{NOTEHEAD_VOID[L:0.6,R:0.9] NOTEHEAD_BLACK[L:0.4,R:0.9]}
INFO  [We Shall Overcome - Orchester - Partitur - SL Edit 1_00a#11]             ReductionStep 105  | Stems free length median value: 37 pixels, 3.4 interlines
INFO  [We Shall Overcome - Orchester - Partitur - SL Edit 1_00a#11]              CueBeamsStep 81   | Step CUE_BEAMS is skipped because small heads switch is off
Estimating resolution as 259
INFO  [We Shall Overcome - Orchester - Partitur - SL Edit 1_00a#11]                      Page 307  | 7 raw measures: [7 in system#1]
INFO  [We Shall Overcome - Orchester - Partitur - SL Edit 1_00a#11]              SlursBuilder 240  | Slurs: 110
INFO  [We Shall Overcome - Orchester - Partitur - SL Edit 1_00a#11]           SegmentsBuilder 155  | Segments: 114
INFO  [We Shall Overcome - Orchester - Partitur - SL Edit 1_00a#11]                  StepMenu 224  | End of sheet step SYMBOLS
INFO  [We Shall Overcome - Orchester - Partitur - SL Edit 1_00a#11]             SentenceInter 461  | No chord near SentenceInter#16691{(0.713) staff:1 mFont:36 Direction} 86
INFO  [We Shall Overcome - Orchester - Partitur - SL Edit 1_00a#11]             SymbolsLinker 258  | No chord linked to OctaveShiftInter#20104{(0.302) staff:1 OTTAVA ALTA}
INFO  [We Shall Overcome - Orchester - Partitur - SL Edit 1_00a#11]                  KeyInter 1165 | No effective key before AlterInter#20123{(0.780) staff:5 NATURAL pitch:3.0}
INFO  [We Shall Overcome - Orchester - Partitur - SL Edit 1_00a#11]                  KeyInter 1165 | No effective key before AlterInter#20124{(0.769) staff:7 NATURAL pitch:-4.0}
INFO  [We Shall Overcome - Orchester - Partitur - SL Edit 1_00a#11]                  KeyInter 1165 | No effective key before AlterInter#20820{(0.795) staff:9 NATURAL pitch:-3.0}
INFO  [We Shall Overcome - Orchester - Partitur - SL Edit 1_00a#11]                  KeyInter 1165 | No effective key before AlterInter#21196{(0.776) staff:14 NATURAL pitch:-4.0}
INFO  [We Shall Overcome - Orchester - Partitur - SL Edit 1_00a#11]                  KeyInter 1165 | No effective key before AlterInter#21224{(0.797) staff:15 NATURAL pitch:1.0}
INFO  [We Shall Overcome - Orchester - Partitur - SL Edit 1_00a#11]                  KeyInter 1165 | No effective key before AlterInter#21225{(0.800) staff:16 NATURAL pitch:1.0}
INFO  [We Shall Overcome - Orchester - Partitur - SL Edit 1_00a#11]                  KeyInter 1165 | No effective key before AlterInter#21194{(0.799) staff:18 NATURAL pitch:-4.0}
WARN  [We Shall Overcome - Orchester - Partitur - SL Edit 1_00a#11]                 SheetStub 1412 | Error in performing [LINKS] java.util.concurrent.ExecutionException: java.lang.RuntimeException: java.lang.NullPointerException: Cannot load from int array because "pitches" is null
java.util.concurrent.ExecutionException: java.lang.RuntimeException: java.lang.NullPointerException: Cannot load from int array because "pitches" is null
	at java.base/java.util.concurrent.FutureTask.report(FutureTask.java:124)
	at java.base/java.util.concurrent.FutureTask.get(FutureTask.java:207)
	at org.audiveris.omr.sheet.SheetStub.doOneStep(SheetStub.java:516)
	at org.audiveris.omr.sheet.SheetStub.reachStep(SheetStub.java:1394)
	at org.audiveris.omr.step.ui.StepMenu$StepAction$1.doInBackground(StepMenu.java:223)
	at org.audiveris.omr.step.ui.StepMenu$StepAction$1.doInBackground(StepMenu.java:172)
	at java.desktop/javax.swing.SwingWorker$1.call(SwingWorker.java:303)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:328)
	at java.desktop/javax.swing.SwingWorker.run(SwingWorker.java:340)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614)
	at java.base/java.lang.Thread.run(Thread.java:1474)
Caused by: java.lang.RuntimeException: java.lang.NullPointerException: Cannot load from int array because "pitches" is null
	at org.audiveris.omr.step.AbstractSystemStep.doitPerSystem(AbstractSystemStep.java:185)
	at org.audiveris.omr.step.AbstractSystemStep.doit(AbstractSystemStep.java:125)
	at org.audiveris.omr.step.OmrStep.doit(OmrStep.java:138)
	at org.audiveris.omr.sheet.SheetStub.lambda$doOneStep$0(SheetStub.java:500)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:328)
	... 3 common frames omitted
Caused by: java.lang.NullPointerException: Cannot load from int array because "pitches" is null
	at org.audiveris.omr.sig.inter.KeyInter.checkConfiguration(KeyInter.java:1184)
	at org.audiveris.omr.sig.inter.KeyInter.lookupCandidates(KeyInter.java:1109)
	at org.audiveris.omr.sheet.symbol.SymbolsLinker.process(SymbolsLinker.java:370)
	at org.audiveris.omr.sheet.symbol.LinksStep.doSystem(LinksStep.java:132)
	at org.audiveris.omr.sheet.symbol.LinksStep.doSystem(LinksStep.java:61)
	at org.audiveris.omr.step.AbstractSystemStep.lambda$doitPerSystem$0(AbstractSystemStep.java:158)
	at org.audiveris.omr.step.AbstractSystemStep.doitPerSystem(AbstractSystemStep.java:178)
	... 7 common frames omitted
INFO  [We Shall Overcome - Orchester - Partitur - SL Edit 1_00a#11]                  StepMenu 224  | End of sheet step SYMBOLS
```

**Audiveris environment:**
```
Audiveris
Role: Optical Music Recognition
Version: 5.9.0:cc4d72640939b5c5ae37ff7af95b441a306fd2c5
Container: file:/app/lib/audiveris.jar
Home page: http://www.audiveris.org
Dev. project: https://github.com/Audiveris/audiveris
Software license: GNU Affero GPL v3
OCR engine: Tesseract OCR, version 5.5.1
OCR folder: /home/wrosner/.var/app/org.audiveris.audiveris/config/tessdata
Java vendor: Flathub
Java version: 25.0.1
Java runtime: OpenJDK Runtime Environment (build 25.0.1)
Java VM: OpenJDK 64-Bit Server VM (build 25.0.1, mixed mode, sharing)
Operating System: Linux 6.12.63+deb13-amd64
Architecture: amd64
```

**Additional context**
Add any other context about the problem here.

```
~$ cat /etc/os-release
PRETTY_NAME="Debian GNU/Linux 13 (trixie)"
NAME="Debian GNU/Linux"
VERSION_ID="13"
VERSION="13 (trixie)"
VERSION_CODENAME=trixie
DEBIAN_VERSION_FULL=13.3
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
```

--- a/app/src/main/java/org/audiveris/omr/sig/inter/KeyInter.java
+++ b/app/src/main/java/org/audiveris/omr/sig/inter/KeyInter.java
@@ -1177,6 +1177,15 @@ public class KeyInter
 
         final ClefKind clefKind = clef.getKind();
 
+        // A key signature is meaningful only for a pitched clef of a known kind.
+        // Percussion clefs (and any clef with no usable kind) carry no key signature and are
+        // absent from the pitches maps. Bail out cleanly here rather than later dereferencing
+        // a null pitches array (see issue #893).
+        if (!SHARP_PITCHES_MAP.containsKey(clefKind)) {
+            logger.debug("No key signature applicable for clef {} before {}", clef, firstAlter);
+            return null;
+        }
+
         Integer cancelFifths = null; // Used only for a cancel key
 
         // Determine target pitches to check vertical position

